Formula/pnpm.rb

```diff
@@ -20,14 +20,79 @@ class Pnpm < Formula
     sha256 cellar: :any_skip_relocation, x86_64_linux:  "e0d669c48801ee9a3f0a49d99f81f2a50dcd1e879526b77a1367c1ae6daa9811"
   end
 
-  depends_on "node"
+  depends_on "node" => :build
+
+  # Described in https://github.com/pnpm/pnpm#installation
+  # Managed by https://github.com/pnpm/get
+  resource "pnpm-buildtime" do
+    url "https://get.pnpm.io/v6.14.js"
+    sha256 "c80817f1dac65ee497fc8ca0b533e497aacfbf951a917ff4652825710bbacda7"
+  end
 
   def install
-    system "npm", "install", *Language::Node.std_npm_install_args(libexec)
-    bin.install_symlink Dir["#{libexec}/bin/*"]
+    (prefix/"etc").mkpath
+    (prefix/"etc/npmrc").atomic_write "global-bin-dir = ${HOME}/Library/pnpm\n" if OS.mac?
+    (prefix/"etc/npmrc").atomic_write "global-bin-dir = ${HOME}/.local/pnpm\n" if OS.linux?
+
+    buildtime_bin = buildpath/"buildtime-bin"
+    resource("pnpm-buildtime").stage do |r|
+      buildtime_bin.install "v#{r.version}.js"
+      (buildtime_bin/"pnpm").write_env_script "#{Formula["node"].bin}/node",
+        buildpath/"buildtime-bin/v#{r.version}.js", {}
+      chmod 0755, buildtime_bin/"pnpm"
+    end
+    ENV.prepend_path "PATH", buildtime_bin
+    system "pnpm", "install"
+    chdir "packages/pnpm" do
+      system "pnpm", "run", "compile"
+    end
+    chdir "packages/beta" do
+      system "node_modules/.bin/pkg", "--target=host", "../pnpm/dist/pnpm.cjs"
+      bin.install "pnpm"
+    end
+  end
+
+  def post_install
+    if OS.linux?
+      executable = bin/"pnpm"
+      unless (system_command executable).success?
+        default_interpreter_size = 27 # /lib64/ld-linux-x86-64.so.2
+        return if executable.interpreter.size <= default_interpreter_size
+
+        offset = PatchELF::Helper::PAGE_SIZE
+        binary = IO.binread executable
+        binary.sub!(/(?<=PAYLOAD_POSITION = ')((\d+) *)(?=')/) do
+          (Regexp.last_match(2).to_i + offset).to_s.ljust(Regexp.last_match(1).size)
+        end
+        binary.sub!(/(?<=PRELUDE_POSITION = ')((\d+) *)(?=')/) do
+          (Regexp.last_match(2).to_i + offset).to_s.ljust(Regexp.last_match(1).size)
+        end
+        executable.atomic_write binary
+      end
+    end
+  end
```

Comment:
What are you trying to do here? Because this looks like a lot of hacks for something that should be fixed upstream.

Reply:
Rewriting the interpreter (/lib64/ld-linux-x86-64.so.2) in ELF causes the problem, and I'm afraid this cannot be fixed upstream because this is a Linuxbrew-specific problem.
If any_skip_relocation prevents the interpreter from being rewritten, then I can remove this hacky code safely.
I don't know why the interpreter is rewritten in Linuxbrew, even if any_skip_relocation is specified.

What I'm doing here is remediation against unnecessary binary patching.

References:
https://github.com/vercel/pkg/issues/321
https://github.com/NixOS/nixpkgs/pull/48193/files#diff-b65952dbe5271c002fbc941b01c3586bf5050ad0e6aa6b2fcc74357680e103ea